### PR TITLE
Gateway handler skeleton

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

File: core/scripts/gateway/workflow/upload_workflow.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+)
+
+func main() {
+	gatewayURL := flag.String("gateway_url", "http://localhost:5002", "Gateway URL")
+	privateKey := flag.String("private_key", "65456ffb8af4a2b93959256a8e04f6f2fe0943579fb3c9c3350593aabb89023f", "Private key to sign the message with")
+	messageID := flag.String("message_id", "12345", "Request ID")
+	methodName := flag.String("method", "add_workflow", "Method name")
+	donID := flag.String("don_id", "workflow_don_1", "DON ID")
+	workflowSpec := flag.String("workflow_spec", "[my spec abcd]", "Workflow Spec")
```
Comment (shileiwill) [inline core/scripts/gateway/workflow/upload_workflow.go:25]: In real world scenario, this workflowSpec is the encoded combo of wasm binary and yaml config?

File: core/scripts/gateway/workflow/upload_workflow.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/joho/godotenv"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+)
+
+func main() {
+	gatewayURL := flag.String("gateway_url", "http://localhost:5002", "Gateway URL")
+	privateKey := flag.String("private_key", "65456ffb8af4a2b93959256a8e04f6f2fe0943579fb3c9c3350593aabb89023f", "Private key to sign the message with")
```
Comment (shileiwill) [inline core/scripts/gateway/workflow/upload_workflow.go:21]: Am I understanding this correctly? this is the workflow owner's private key and some addresses/private keys are allowlisted in the Gateway DON.

Comment (bolekk) [reply core/scripts/gateway/workflow/upload_workflow.go:21]: Yes - owner's private key used to sign messages to Gateway.
Yes - there needs to be some kind of allowlist or an account system on Gateway's side (of addresses, not private keys).

File: core/scripts/gateway/workflow/upload_workflow.go (hunk 1 above)
Comment (bolekk) [reply core/scripts/gateway/workflow/upload_workflow.go:25]: Yes.

**Pull request closed without merging**